src/org/geekhub/json/JsonSerializer.java

```diff
@@ -70,7 +74,28 @@ public static Object serialize(Object o) {
      * @throws InstantiationException
      */
     private static JSONObject toJsonObject(Object o) throws Exception {
-        //implement me
-        return null;
+        Class<?> objectClass = o.getClass();
+        JSONObject jsonObject = new JSONObject();
+        for(Field field : objectClass.getDeclaredFields()){
+            field.setAccessible(true);
+
+            if (null != field.getAnnotation(Ignore.class)) {
+                continue;
+
+            } else if (simpleTypes.contains(field.getType())) {
+                jsonObject.put(field.getName(), field.get(o));
+
+            } else if (null != field.getAnnotation(UseDataAdapter.class)) {
+                Class<JsonDataAdapter> dataAdapter = (Class<JsonDataAdapter>)field
```

Comment:
Remove generics and casting. Put instance of JsonDataAdapter to dataAdapter variable and just call to json in next line.